### PR TITLE
fix up Haml::Compiler extension so it works on both Haml 3.1 and 3.2

### PR DESCRIPTION
**Pull request by jacquescrocker**
Allows hamlbars to work on both Haml 3.1 and 3.2

### CODE REVIEW

**Comment by jimsynz** [comment]
THANKS FOR THE CONTRIBUTION YOU ARE AWESOME

File: spec/template_spec.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@
   it "should bind element attributes" do
     template_file.write('%img{ :bind => { :src => "logoUri" }, :alt => "Logo" }')
     template_file.rewind
-    template = Hamlbars::Template.new(template_file)
+    template = Hamlbars::Template.new(template_file, :format => :xhtml)
```
Comment (joliss) [inline spec/template_spec.rb:34]: @railsjedi: I have a test refactoring that I'm rebasing on top of this, so could you explain for me what the `:format => :xhtml` is testing please?

Comment (joliss) [inline spec/template_spec.rb:34]: In other words, can I drop it without losing coverage on anything important?

Comment (jacquescrocker) [inline spec/template_spec.rb:34]: Haml 3.2 switches the default to html5 output. Explicitly declareing `:format => :xhtml` will make sure things stay the same between 3.1 and 3.2

Comment (joliss) [inline spec/template_spec.rb:34]: Cool, thanks.

**Pull request merged**